beets: modernize and disable failing test

GaetanLepage opened the pull request:
## Things done

cc @aszlig @doronbehar @lovesegfault @montchr @pjones

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
beets: modernize

doronbehar commented on pkgs/tools/audio/beets/common.nix:
```diff
@@ -158,6 +161,8 @@ python3Packages.buildPythonApplication {
     "test/plugins/test_autobpm.py::TestAutoBPMPlugin::test_import"
     # AssertionError: assert 0 == 117
     "test/plugins/test_autobpm.py::TestAutoBPMPlugin::test_command"
+    # AssertionError: Image written is not /build/source/test/rsrc/abbey-similar.jpg
+    "test_accept_similar_art"
```
Could we separate this build failure fix to a separate commit? Also note https://github.com/beetbox/beets/pull/5650 and https://github.com/NixOS/nixpkgs/issues/385389 . I'd like to wait a day or so for upstream to merge that PR.

doronbehar commented on pkgs/tools/audio/beets/common.nix:
```diff
@@ -137,14 +138,16 @@ python3Packages.buildPythonApplication {
       rarfile
       responses
-    ]
+    ])
     ++ pluginWrapperBins;
```
Nit: The `writeableTmpDirAsHomeHook` addition (nice to know it exists!) can be added here, and the diff might get slightly smaller and nicer :).

doronbehar reviewed (commented):
Thanks for the modernization bits :).

GaetanLepage committed:
beets: fetch patch to fix IMBackend

GaetanLepage committed:
beets: disable flaky tests on darwin

GaetanLepage committed:
beetsPackages.alternatives: modernize

GaetanLepage committed:
beetsPackages.alternatives: 0.13.0 -> 0.13.1

Diff: https://github.com/geigerzaehler/beets-alternatives/compare/refs/tags/v0.13.0...v0.13.1

Changelog: https://github.com/geigerzaehler/beets-alternatives/blob/v0.13.1/CHANGELOG.md

doronbehar commented on pkgs/tools/audio/beets/common.nix:
```diff
@@ -144,6 +145,7 @@ python3Packages.buildPythonApplication {
       mock
       rarfile
       responses
+      writableTmpDirAsHomeHook
     ]
```
Personally I don't mind that much the fact a non Python package is mixed here, but I was thinking of:

```suggestion
    ] ++ [
      writeableTmpDirAsHomeHook
    ]

doronbehar commented on pkgs/tools/audio/beets/plugins/alternatives.nix:
```diff
@@ -13,28 +14,30 @@ python3Packages.buildPythonApplication rec {
+    pytest-cov-stub
     mock
     typeguard
+    writableTmpDirAsHomeHook
```
Same here regarding the mix of Python & non-Python inputs.

doronbehar commented on pkgs/tools/audio/beets/plugins/alternatives.nix:
```diff
@@ -43,5 +43,8 @@ python3Packages.buildPythonApplication rec {
       lovesegfault
     ];
     license = lib.licenses.mit;
+    # Tests fail on darwin with:
+    # Exception: Unsupported system: Darwin
+    platforms = lib.platforms.linux;
```
In version 0.13.1 this shouldn't fail. See:

https://github.com/geigerzaehler/beets-alternatives/compare/v0.13.0..v0.13.1#diff-4dd682d5c858fe7d3f39400e3da7a0ea9e77232bab93c6d51e5cb523844665b2L248

doronbehar reviewed (commented):
Looks pretty good overall. Thanks for taking care of these issues!

doronbehar commented on pkgs/tools/audio/beets/common.nix:
```diff
@@ -162,6 +162,9 @@ python3Packages.buildPythonApplication {
     "test/plugins/test_autobpm.py::TestAutoBPMPlugin::test_import"
     # AssertionError: assert 0 == 117
     "test/plugins/test_autobpm.py::TestAutoBPMPlugin::test_command"
+    # AssertionError: Image written is not /build/source/test/rsrc/abbey-similar.jpg
+    # https://github.com/beetbox/beets/pull/5650
+    "test_accept_similar_art"
```
You know, I'd really prefer a patch here, as upstream's PR is definitely fixing this (I tested), and with a patch, on the next version bump we'd remove the patch and we'd the build will naturally continue to run this test; Whereas, with this disable the next version bumper might miss the fact this PR was merged and hence the test can be enabled again... Let's give upstream a few hours at least :pray: ? I don't see any reason they shouldn't merge that PR and hence provide us with a stable link to a patch..

doronbehar replied:
They merged it just now.

doronbehar commented:
BTW, how do you produce these 1 comment Nixpkgs review reports? Do you aggregate the `report.md` files contents from different machines every time?
